gateway/gateway.go

```diff
@@ -563,51 +561,85 @@ func (gateway *HandleT) getJobDataFromRequest(req *webRequestT) (jobData *jobFro
 	}
 
 	// set anonymousId if not set in payload
-	result := gjson.GetBytes(body, "batch")
-	var out []map[string]interface{}
-	var builtUserID string
-	var notIdentifiable, nonRudderEvent, containsAudienceList bool
-	result.ForEach(func(_, vjson gjson.Result) bool {
-		anonIDFromReq := strings.TrimSpace(vjson.Get("anonymousId").String())
-		userIDFromReq := strings.TrimSpace(vjson.Get("userId").String())
-		if builtUserID == "" {
-			if anonIDFromReq != "" {
-				builtUserID = userIDHeader + DELIMITER + anonIDFromReq + DELIMITER + userIDFromReq
-			} else {
-				// Proxy gets the userID from body if there is no anonymousId in body/header
-				builtUserID = userIDHeader + DELIMITER + userIDFromReq + DELIMITER + userIDFromReq
-			}
-		}
+	var (
+		out                                                   []map[string]interface{}
+		builtUserID, userIDFromReq                            string
+		sourcesJobRunID, sourcesTaskRunID                     string
+		sdkName, sdkVersion                                   string
+		notIdentifiable, nonRudderEvent, containsAudienceList bool
+	)
 
-		eventTypeFromReq := strings.TrimSpace(vjson.Get("type").String())
+	lo.ForEach(eventsBatch, func(vjson gjson.Result, _ int) {
+		anonIDFromReq := strings.TrimSpace(vjson.Get("anonymousId").String())
+		userIDFromReq = strings.TrimSpace(vjson.Get("userId").String())
```

Comment:
We're checking these two for each events in the batch but then we're considering only the last value of `userIDFromReq` when checking later for suppressions (i.e. `if gateway.isUserSuppressed(workspaceId, userIDFromReq, sourceID) {`)?

Is this correct?

Reply:
the earlier logic was something similar, differing only in that the `userID` from the first event of the batch was being considered when looking up suppression.
I'm making some more changes around it.
Should we consider looking up supressions for all the users in a batch..?

Reply:
brought back looking if userID from first event of the batch is suppressed behaviour.
> Should we consider looking up suppressions for all the users in a batch..?

this question still stands though

Reply:
Apparently it [might be possible](https://rudderlabs.slack.com/archives/C01HTT66UMB/p1675240518176459?thread_ts=1675239804.575179&cid=C01HTT66UMB) to get multiple users in the same batch.

It would be correct to filter all the filterable ones and then work with whatever batch is left. But let's address this in another PR :+1: